### PR TITLE
Remove the LabelSet parameter on the DoubleCounter.

### PR DESCRIPTION
**Pull request by jkwatson**
@bogdandrutu does this look like an ok start to this?

related to #992

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/internal/Utils.java [hunk 1]
```diff
@@ -208,4 +208,23 @@ private static String format(String template, @javax.annotation.Nullable Object.
...
+  /**
+   * Validates that the array of Strings is 1) even in length, and 2) they can be formed into valid
+   * pairs where the first item in the pair is not null.
+   *
+   * <p>TODO: write unit tests for this method.
+   *
+   * @param keyValuePairs The String[] to validate for correctness.
+   * @throws IllegalArgumentException if any of the preconditions are violated.
+   */
+  public static void validateLabelPairs(String[] keyValuePairs) {
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/internal/Utils.java]: nit: Do we need to allow null values? null means you don't add the key to the list.

File: api/src/main/java/io/opentelemetry/metrics/DoubleCounter.java [hunk 2]
```diff
@@ -61,10 +61,10 @@
    * <p>The value added is associated with the current {@code Context} and provided LabelSet.
    *
    * @param delta the value to add.
-   * @param labelSet the labels to be associated to this recording
+   * @param labelKeyValuePairs the labels to be associated to this recording.
    * @since 0.1.0
    */
-  void add(double delta, LabelSet labelSet);
+  void add(double delta, String... labelKeyValuePairs);
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/DoubleCounter.java]: Bind also needs to change.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DoubleCounter.java]: yep. I was going to do the bind usage as the next PR

File: api/src/main/java/io/opentelemetry/internal/Utils.java (hunk 1 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/internal/Utils.java]: not sure. we could ignore null keys, I suppose? Want to create an issue to track it?

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/internal/Utils.java]: Let's keep it as it is. We should just create an issue about how to pass Labels and track what we discussed on gitter  and we will address that concern at that point.

**Comment by bogdandrutu** [comment]
@jkwatson I am happy if you change in this PR all direct calls (add, record).

**Comment by jkwatson** [comment]
> @jkwatson I am happy if you change in this PR all direct calls (add, record).

ok! giant PR with all the changes for all the instrumented coming soon!  (i kid, I'll just do DoubleCounter for this one )

**Comment by jkwatson** [comment]
@bogdandrutu updated. It ended up bigger than I'd like, since the `bind` was on the Instrument interface and adding it ended up trickling down into all the instruments.

File: api/src/main/java/io/opentelemetry/internal/Utils.java (hunk 1 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/internal/Utils.java]: #1000

**Comment by jkwatson** [comment]
oh shoot. You wanted to change all the direct call and I did all the bound calls.  :(

**Comment by jkwatson** [comment]
closing in favor of #1001

**Pull request closed without merging**